tests/acceptance/dex/idtokenprovider.go

```diff
@@ -0,0 +1,116 @@
+package dex
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"log"
+
+	"github.com/pkg/errors"
+)
+
+type idTokenProvider interface {
+	fetchIdToken() (string, error)
+}
+
+type dexIdTokenProvider struct {
+	httpClient *http.Client
+	config     idProviderConfig
+}
+
+func newDexIdTokenProvider(httpClient *http.Client, config idProviderConfig) idTokenProvider {
+	return &dexIdTokenProvider{httpClient, config}
+}
+
+func (p *dexIdTokenProvider) fetchIdToken() (string, error) {
+
+	flowResult, err := p.implicitFlow()
+	if err != nil {
+		return "", err
+	}
+	return flowResult["id_token"], nil
+}
+
+func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
+
+	authorizeResp, err1 := p.httpClient.PostForm(p.config.dexConfig.authorizeEndpoint, url.Values{
```

Comment:
use one error variable - `err` and reassign the error there

Reply:
done